pkg/ssh/forward.go

```diff
@@ -86,3 +97,85 @@ func forward(localConn net.Conn, client *ssh.Client, remoteNetwork, remoteAddr s
 	waitGroup.Add(1)
 	waitGroup.Wait()
 }
+
+func ReversePortForward(
+	ctx context.Context,
+	client *ssh.Client,
+	remoteNetwork, remoteAddr, localNetwork, localAddr string,
+	exitAfterTimeout time.Duration,
+	log log.Logger,
+) error {
+	listener, err := client.Listen(remoteNetwork, remoteAddr)
+	if err != nil {
+		return err
+	}
+	defer listener.Close()
+
+	done := make(chan struct{})
+	defer close(done)
+
+	go func() {
+		select {
+		case <-done:
+		case <-ctx.Done():
+			_ = listener.Close()
```

Comment:
this function looks exactly the same as `PortForward()` - only differs in calling `forward()` and `reverseForward()`. 

What about not duplicating the code, by reuse it?

Reply:
And those two (forward/reverseForward()) also look similar. Maybe we don't need to duplicate that code as well.

Reply:
Yep done now for Forward/ReverseForward

But the actual forwarding functions (forward/reverse) have quite separate concepts and connections, so I'd like to keep them separated for the sake of simplicity